### PR TITLE
Feat branch/google integration

### PR DESCRIPTION
**Pull request by acrojas2**
### Contexto
​Lakatan quiere integrarse con GoogleAdmin. para poder manejar los usuarios de una organizacion y asi poder agregar y quitar usuarios
### Qué se esta haciendo
Se agregan los flujos de happy path para poder crear un usuario de google en lakatan

​
#### En particular hay que revisar
​

-----------

### CODE REVIEW

File: engines/googleauthentication/app/controllers/googleauthentication/api/v1/authentications_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Googleauthentication::Api::V1::AuthenticationsController < Googleauthentication::Api::V1::BaseController
```
Comment (monkeyci) [inline engines/googleauthentication/app/controllers/googleauthentication/api/v1/authentications_controller.rb:1]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/LineLength: Line is too long. [110/100]

File: engines/googleuseraccounts/app/controllers/googleuseraccounts/api/v1/google_user_accounts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Googleuseraccounts::Api::V1::GoogleUserAccountsController < Googleuseraccounts::Api::V1::BaseController
```
Comment (monkeyci) [inline engines/googleuseraccounts/app/controllers/googleuseraccounts/api/v1/google_user_accounts_controller.rb:1]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/LineLength: Line is too long. [109/100]

File: engines/googleauthentication/app/assets/config/googleauthentication_manifest.js [hunk 3]
```diff
@@ -0,0 +1 @@
+//= link_directory ../stylesheets/googleauthentication .css
```
Comment (monkeyci) [inline engines/googleauthentication/app/assets/config/googleauthentication_manifest.js:1]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expected space or tab after '//' in comment  spaced-comment

File: engines/googleuseraccounts/app/assets/config/googleuseraccounts_manifest.js [hunk 4]
```diff
@@ -0,0 +1 @@
+//= link_directory ../stylesheets/googleuseraccounts .css
```
Comment (monkeyci) [inline engines/googleuseraccounts/app/assets/config/googleuseraccounts_manifest.js:1]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expected space or tab after '//' in comment  spaced-comment

File: engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
+module Googleauthentication
+  class GoogleAuthStore
+    def initialize(integration_name)
+      @integration_name = integration_name
```
Comment (monkeyci) [inline engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb:4]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 4

File: engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    end
+
+    def load(organization_id)
+      google_token(organization_id).value
```
Comment (monkeyci) [inline engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb:8]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 8

File: engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    end
+
+    def store(organization_id, token)
+      Googleauthentication::GoogleToken.find_or_create_by!(
```
Comment (monkeyci) [inline engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb:12]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 12

File: engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        integration_name: @integration_name,
+        organization_configuration_id: organization_configuration(organization_id).id
+      ) do |google_token|
+        google_token.value = token
```
Comment (monkeyci) [inline engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb:16]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 16

File: engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      end
+    end
+
+    def delete(organization_id)
+      google_token(organization_id).update!(value: nil)
```
Comment (monkeyci) [inline engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb:21]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 21

File: engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    end
+
+    private
+
+    def google_token(organization_id)
+      Googleauthentication::GoogleToken.joins(:organization_configuration).find_by!(
```
Comment (monkeyci) [inline engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb:27]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 27

File: engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        integration_name: @integration_name,
+        'organization_configurations.organization_id': organization_id
+      )
+    end
+
+    def organization_configuration(organization_id)
+      @organization_configuration ||= Googleauthentication.organization_configuration_model
```
Comment (monkeyci) [inline engines/googleauthentication/app/stores/googleauthentication/google_auth_store.rb:34]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 34

File: engines/googleauthentication/lib/googleauthentication.rb [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def self.redirect_url
+    "#{ENV['APPLICATION_HOST']}/oauth2"
+  end
+
+  class << self
+    MODULE_DEPENDENCIES.each do |symbol|
+      define_method(symbol) do
+        class_variable_get(:"@@#{symbol}").constantize
```
Comment (monkeyci) [inline engines/googleauthentication/lib/googleauthentication.rb:25]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 25

File: engines/googleuseraccounts/lib/googleuseraccounts.rb [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+  mattr_accessor *MODULE_DEPENDENCIES
+
+  def configure
+    yield self
+    require "googleuseraccounts"
+  end
+
+  class << self
+    MODULE_DEPENDENCIES.each do |symbol|
+      define_method(symbol) do
```
Comment (monkeyci) [inline engines/googleuseraccounts/lib/googleuseraccounts.rb:17]: <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Not covered lines: 17 and 18

**Review by aaguilarg — APPROVED** [review]

**Pull request merged**